oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/LogoutService.java

```diff
@@ -155,4 +167,30 @@ public APIGatewayProxyResponseEvent generateLogoutResponse(
         return generateApiGatewayProxyResponse(
                 302, "", Map.of(ResponseHeaders.LOCATION, uri.toString()), null);
     }
+
+    private APIGatewayProxyResponseEvent generateAccountInterventionLogoutResponse(
+            APIGatewayProxyRequestEvent input,
+            Optional<String> clientId,
+            Optional<String> sessionId,
+            AccountInterventionStatus accountStatus) {
+        String redirectUri;
+        if (accountStatus.blocked()) {
+            redirectUri = configurationService.getAccountStatusBlockedURI();
+            LOG.info("Generating Account Intervention blocked logout response");
+        } else if (accountStatus.suspended()) {
+            redirectUri = configurationService.getAccountStatusSuspendedURI();
+            LOG.info("Generating Account Intervention suspended logout response");
+        } else {
+            throw new RuntimeException("Account status must be blocked or suspended");
+        }
+        sessionId.ifPresent(t -> cloudwatchMetricsService.incrementLogout(clientId));
```

Comment:
I know this line has come from elsewhere, but it seems like an odd check to make. Do you understand it?

Reply:
I [added](https://github.com/govuk-one-login/authentication-api/pull/3206) Cloudwatch metrics for the LogoutHandler a few months ago that didn't check for the `sessionId`. [This PR](https://github.com/govuk-one-login/authentication-api/pull/3248) adds the `sessionId` check, presumably after feedback once the metric was live .